packages/http-client-java/emitter/src/code-model-builder.ts

```diff
@@ -1076,21 +1066,29 @@ export class CodeModelBuilder {
       }
       return undefined;
     }
-    function getLastSegmentSerializedName(
-      segments: SdkModelPropertyType[] | undefined,
-    ): string | undefined {
-      const lastSegment = getLastPropertySegment(segments);
-      return lastSegment ? getPropertySerializedName(lastSegment) : undefined;
-    }
 
-    // TODO: in future the property could be nested, so that the "itemSegments" or "nextLinkSegments" would contain more than 1 element
-    // item/result
-    // "itemsSegments" should exist for "paging"/"lropaging"
-    const itemSerializedName = getLastSegmentSerializedName(sdkMethod.response.resultSegments);
+    // pageItems
+    const pageItemsResponseProperty = findResponsePropertySegments(
+      op,
+      sdkMethod.response.resultSegments,
+    );
+    // "sdkMethod.response.resultSegments" should not be empty for "paging"/"lropaging"
+    // "itemSerializedName" take 1st property for backward compatibility
+    const itemSerializedName =
+      pageItemsResponseProperty && pageItemsResponseProperty.length > 0
+        ? pageItemsResponseProperty[0].serializedName
+        : undefined;
+
     // nextLink
-    const nextLinkSerializedName = getLastSegmentSerializedName(
+    const nextLinkResponseProperty = findResponsePropertySegments(
+      op,
       sdkMethod.pagingMetadata.nextLinkSegments,
     );
+    // "nextLinkSerializedName" take 1st property for backward compatibility
```

Comment:
Could you please show what does output code-model.yaml look like?

Reply:
This be same, e.g.
```
nextLinkSerializedName: nestedNext
```
(it can take the first or the last, but if codegen only work based on this, the generated code will not compile -- the type would not match the ctor of `PagedResponse` class)

The new `pageItemsResponseProperty` gives the full path of property, on `nestedNext` -> `next`

Reply:
I  see, so we support multiple level nested cases as well?

Reply:
Yeah, I mean nested -> nested -> nested (more than one level nested) cases.

Reply:
Yeah, the PR support arbitrary length of such nested properties (basically there is a for-loop on Mapper and Template).